src/setup_payload/ManualSetupPayloadParser.h

```diff
@@ -29,6 +29,12 @@
 
 namespace chip {
 
+CHIP_ERROR CheckDecimalStringValidity(std::string decimalString, std::string & decimalStringWithoutCheckDigit);
+CHIP_ERROR CheckCodeLengthValidity(const std::string & decimalString, bool isLongCode);
+CHIP_ERROR ToNumber(const std::string & decimalString, uint32_t & dest);
+CHIP_ERROR ReadDigitsFromDecimalString(const std::string & decimalString, size_t & index, uint32_t & dest,
+                                       size_t numberOfCharsToRead);
```

Comment:
Those are ... very specific functions to have in a global `chip` namespace and be exposing.

Are we just exposing them because the tests call them?  If so, the right thing to do is probably to condition the exposure on `CHIP_CONFIG_TEST` at the very least, if not something that is set only when compiling the relevant tests (e.g. like `CHIP_ENABLE_TEST_ENCRYPTED_BUFFER_API`).  And when they are exposed they should be in a more narrowly tailored namespace than just `::chip`.  Or should become static methods of the ManualSetupPayloadParser class, perhaps?

Reply:
This is a preexisting issue and this PR is actually a step forward. These functions are used in more than one translation unit.

Before multiple files were including the .cpp file that defines them which is much worse than this (and causes ODR problems for the folks trying to link the setup_payload tests into a single binary which is why it is changed here).

Reply:
I think adding such ifdefs is a bad direction, and it is not very forward looking (incompatible with C++ modules).

These functions look pretty simple, maybe we could move them to their own file and mark that file's header private (but it won't stop someone who's determined and it won't help for build systems that don't have such an encapsulation primitive).

Reply:
> Before multiple files were including the .cpp file that defines them

I agree this was terrible.  The right thing to do is to make them class statics, not file statics, and then there is no problem at least with the namespacing....  If we then want to make them protected by default with the tests a friend, we can do that too...  But the most important part for me is the namespace pollution that we should not have.

(The reason it would be good to hide these functions is that no one except this one class and its tests should ever really need them.  Better to not have easily-misused things lying around.)

Reply:
I changed files exposure from file static to class static. I hope it's a good change :)